lib/liquidTestRunner.js

```diff
@@ -297,30 +307,78 @@ async function deleteExistingHTMLs() {
   } catch (err) {}
 }
 
-async function handleHTMLfiles(testName = "", testRun) {
+async function handleHTMLfiles(testName = "", testRun, mode) {
   deleteExistingHTMLs();
+
+  const getHTMLrenders = async (mode, testName, testRun, openBrowser) => {
+    switch (mode) {
+      case "all":
+        await getHTML(
+          testRun.tests[testName].html_input,
+          testName,
+          openBrowser,
+          "html_input"
+        );
+        await getHTML(
+          testRun.tests[testName].html_preview,
+          testName,
+          openBrowser,
+          "html_preview"
+        );
+        break;
+      case "input":
+        await getHTML(
+          testRun.tests[testName].html_input,
+          testName,
+          openBrowser,
+          "html_input"
+        );
+        break;
+      case "preview":
+        await getHTML(
+          testRun.tests[testName].html_preview,
+          testName,
+          openBrowser,
+          "html_preview"
+        );
+        break;
+    }
+  };
+
   if (testName) {
     // Only one test
-    await getHTML(testRun.tests[testName].html, testName, true);
+    getHTMLrenders(mode, testName, testRun, true);
   } else {
     // All tests
     const testNames = Object.keys(testRun.tests);
     testNames.forEach(async (testName) => {
-      await getHTML(testRun.tests[testName].html, testName, true);
+      getHTMLrenders(mode, testName, testRun, true);
     });
   }
 }
 
 // Used by VSCode Extension
-async function runTests(firmId, handle, testName = "", html_render = false) {
+async function runTests(firmId, handle, testName = "", previewOnly, mode) {
```

Comment:
I think we should approach this differently, since there is no point of having `previewOnly` as `true` when `mode` is `none` ?
I think we should do those check outside the function and pass the correct option (so, `mode` only actually would have all the information) -> and rename it to `htmlMode` or somethimg more clear

Reply:
I'm not sure I follow, `previewOnly` is also a flag they can add to the command (`--preview-only`) and doesn't have any direct link with `mode`.

They can idd still add no HTML render option and `--preview-only` flag, so in that case, the flag would be redundant since mode is already `none`, but I didn't really see that as an issue? 

This is the fastest option I could think of on how to cover the case where want to render HTML without any liquid testing results, but also still give the user the option to combine both. 

Cause now I wanted to cover all these use-cases:
* Get liquid test results only: use `run-test` command without any flag
* Get liquid test results + HTML render: use `run-test` with `--html-input` and/or `--html-preview` flag
* Get HTML render only: use `run-test` with `--html-input` and/or `--html-preview` flag and the `--preview-only` flag. 

Unless you see a better way of tackling these use-cases?

Reply:
Related to this comment: https://github.com/silverfin/silverfin-cli/pull/55#discussion_r1322846486. 
We can have a quick chat if it's not clear.
tl;dr: `--preview-only` for `run-tests` or `--html` for `development-mode` raise an error for me
i still think it's a bit confusing too many options an parameters and we can do it before calling the different run tests functions, but not really required.

Reply:
The only way I could think about on how to make it more clear is if we also do a bigger change in the CLI commands and split run-tests / render into separate processes.